### PR TITLE
[inductor] Enable mypy checking for torch/_inductor/bounds.py

### PR DESCRIPTION
**Pull request by masnesral**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #109271

Summary: Add type hints and enable mypy checking for torch/_inductor/bounds.py

Test Plan: lintrunner

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

### CODE REVIEW

File: torch/_inductor/bounds.py [hunk 1]
```diff
@@ -54,8 +56,10 @@ def get_bounds(self):
             interpreter.run(V.get_ops_handler(), initial_env=self._bounds)
         return self._bounds
 
-    def swap_submodules(self, submodules):
-        result = {}
+    def swap_submodules(
+        self, submodules: Dict[str, Callable[..., Any]]
+    ) -> Dict[str, Callable[..., Any]]:
+        result: Dict[str, Callable[..., Any]] = {}
```
Comment (lezcano) [inline torch/_inductor/bounds.py:62]:
```suggestion
    ) -> Dict[str, Callable[..., ValueRanges]]:
        result: Dict[str, Callable[..., ValueRanges]] = {}
```

File: torch/_inductor/bounds.py [hunk 2]
```diff
@@ -75,7 +79,14 @@ def swap_submodules(self, submodules):
...
 
-    def masked_subblock(self, subblock, env, mask, value, submodules):
+    def masked_subblock(
+        self,
+        subblock: LoopBodyBlock,
+        env: Dict[torch.fx.Node, ValueRanges],
+        mask: Any,
+        value: Any,
+        submodules: Dict[str, Callable[..., Any]],
+    ):
```
Comment (lezcano) [inline torch/_inductor/bounds.py:89]:
```suggestion
    ) -> ValueRanges:
```

**Review by lezcano — COMMENTED** [review]
Some of these annotations are very informative, but others are... uf. Not a big fan of those others, but well.

Just two small things.

**Review by lezcano — APPROVED** [review]

File: torch/_inductor/bounds.py (hunk 2 above)
Comment (masnesral) [reply torch/_inductor/bounds.py:89]: > Some of these annotations are very informative, but others are... uf. Not a big fan of those others, but well.

@lezcano Are you commenting about the state of the types effectively already in use here or do you mean we could do better, e.g., get more detailes with the Callables? If you think we can do better, I'm up for trying harder here. I _think_ many of these were what I gathered via monkeytype, but I could certainly try again.

Comment (lezcano) [reply torch/_inductor/bounds.py:89]: Oh, no, it's not your fault, it's just that many of these are rather verbose and they "do not add much information" (I'm looking at you `Callable[..., Any]`. Nothing actionable though.

I'm sorry if it sounded the wrong way :)

Comment (masnesral) [reply torch/_inductor/bounds.py:89]: Oh no problem at all, just wanted to make sure I understood!

**Comment by masnesral** [comment]
For the test failures, I see: 

> ModuleNotFoundError: No module named 'sympy'

@lezcano , any idea what that's about?

**Comment by lezcano** [comment]
I think that error is not there anymore, so let's try to merge this again

**Pull request closed without merging**